examples/bunYoutubeScraper/src/bunYoutubeScraper.js

```diff
@@ -0,0 +1,55 @@
+const { write } = Bun;
+const BunYoutubeScraper = async (url, saveJson) => {
+  // Make a request to the url
+  const response = await fetch(url);
+  // if response is not ok, throw error
+  if (!response.ok) throw Error('Scrape shield encountered!');
+  // Save content of the response in a variable.
+  const tags = [];
+  // List of tags to scrape.
+  const tagsList = [
+    'meta[name="title"]',
+    'meta[property="og:image"]',
+    'meta[name="description"]',
+    'meta[name="keywords"]',
+    'meta[itemprop="datePublished"]',
+    'meta[itemprop="uploadDate"]',
+    'meta[itemprop="videoId"]',
+  ]
+  // GET the HTML of the page and push it to the tags array.
+  const html = new HTMLRewriter()
+    .on(tagsList, {
+      element(el) {
+        tags.push(el.getAttribute('content')); 
+      }
+    }).transform(response).text()
+  const title = tags[0]
+  const keywords = tags[2]
+  const description = tags[1]
+  const videoThumbnai = tags[3]
+  const videoId = tags[4]
+  const uploadDate = tags[5]
+  const datePublished = tags[6]
+  const viewCount = await html.then(text => text.match(/{"viewCount":{"simpleText":"(.*?)"/g)[0].split("\"")[5])
```

Comment:
Let's use `HTMLRewriter` for all of these. Then it's a Bun HTMLRewriter + YouTube example. You can chain multiple `on()` handlers, sort of like jQuery.